File: ko-nomad/src/types/city.ts
export interface CityScore {
  cafeWork: number;
  coworking: number;
  internet: number;
  costSatisfaction: number;
  transport: number;
  delivery: number;
  medical: number;
  nature: number;
  noise: number;
  airQuality: number;
  safety: number;
  nightlife: number;
  community: number;
  foreignFriendly: number;
}

export interface CityData {
  slug: string;
  name: string;
  region: string;
  regionGroup: "수도권" | "강원" | "충청" | "전라" | "경상" | "제주";
  environment: "해변" | "산" | "도심" | "소도시";
  image: string;
  description: string;
  monthlyCost: number;
  rentCost: number;
  foodCost: number;
  transportCost: number;
  internetSpeed: number;
  cafeCount: number;
  coworkingCount: number;
  temperature: number;
  weatherIcon: string;
  airQualityIndex: number;
  scores: CityScore;
  nomadScore: number;
  reviewCount: number;
  likePercent: number;
}

export interface Review {
  id: string;
  citySlug: string;
  userName: string;
  userImage: string;
  stayDuration: "1주 미만" | "1~4주" | "1~3개월" | "3개월+";
  stayPurpose: "원격근무" | "프리랜서" | "워케이션" | "기타";
  visitDate: string;
  scores: {
    cafeWork: number;
    internet: number;
    cost: number;
    transport: number;
    housing: number;
    nature: number;
    safety: number;
    community: number;
  };
  pros: string;
  cons: string;
  totalScore: number;
  helpful: number;
  createdAt: string;
}

export type SortOption =
  | "nomadScore"
  | "cost_asc"
  | "cost_desc"
  | "internet"
  | "reviewCount"
  | "likePercent";

export type RegionGroup = "전체" | "수도권" | "강원" | "충청" | "전라" | "경상" | "제주";
export type EnvironmentType = "전체" | "해변" | "산" | "도심" | "소도시";

File: ko-nomad-api/src/main/kotlin/com/konomad/entity/City.kt
package com.konomad.entity

import jakarta.persistence.*

@Entity
@Table(name = "cities")
data class City(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(unique = true, nullable = false)
    val slug: String = "",

    val name: String = "",

    val region: String = "",

    val regionGroup: String = "",

    val environment: String = "",

    val image: String = "",

    @Column(length = 500)
    val description: String = "",

    val monthlyCost: Int = 0,

    val rentCost: Int = 0,

    val foodCost: Int = 0,

    val transportCost: Int = 0,

    val internetSpeed: Int = 0,

    val cafeCount: Int = 0,

    val coworkingCount: Int = 0,

    val temperature: Int = 0,

    val weatherIcon: String = "",

    val airQualityIndex: Int = 0,

    val scoreCafeWork: Int = 0,

    val scoreCoworking: Int = 0,

    val scoreInternet: Int = 0,

    val scoreCostSatisfaction: Int = 0,

    val scoreTransport: Int = 0,

    val scoreDelivery: Int = 0,

    val scoreMedical: Int = 0,

    val scoreNature: Int = 0,

    val scoreNoise: Int = 0,

    val scoreAirQuality: Int = 0,

    val scoreSafety: Int = 0,

    val scoreNightlife: Int = 0,

    val scoreCommunity: Int = 0,

    val scoreForeignFriendly: Int = 0,

    val nomadScore: Double = 0.0,

    val reviewCount: Int = 0,

    val likePercent: Int = 0
)
